src/adapter/src/optimize/copy_to.rs

```diff
@@ -300,6 +300,25 @@ impl<'s> Optimize<LocalMirPlan<Resolved<'s>>> for Optimizer {
         // Set the `as_of` and `until` timestamps for the dataflow.
         df_desc.set_as_of(timestamp_ctx.antichain());
 
+        // Get the single timestamp representing the `as_of` time.
+        let as_of = df_desc
+            .as_of
+            .clone()
+            .expect("as_of antichain")
+            .into_option()
+            .expect("unique as_of element");
+
+        // Resolve all unmaterializable function calls including mz_now().
+        let style = ExprPrepStyle::OneShot {
```

Comment:
Could it be the case that `prep_relation_expr` or `prep_scalar_expr` could care about `until`? If so, we should move the following block before this.

Reply:
Fortunately, that can't happen, because we don't pass the `DataflowDescription` to `prep_relation_expr` or `prep_scalar_expr`, so they have no way of observing what `until` we have set on the `DataflowDescription`.